app/uk/gov/hmrc/tai/cacheResolver/estimatedPay/UpdatedEstimatedPayJourneyCache.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tai.cacheResolver.estimatedPay
+
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.tai.service.JourneyCacheService
+import uk.gov.hmrc.tai.util.JourneyCacheConstants
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait UpdatedEstimatedPayJourneyCache extends JourneyCacheConstants{
+
+  def journeyCache(key: String, cacheMap: Map[String, String])
+                  (implicit hc: HeaderCarrier, ec: ExecutionContext, journeyCacheService: JourneyCacheService): Future[Map[String, String]] = {
+
+    def updateCache(cacheToUpdate: Map[String,String],keysToEmpty: List[String])(implicit hc: HeaderCarrier): Future[Map[String, String]] = {
+      if (cacheMap(key) == "Yes") {
```

Comment:
This conditional doesn't make it clear why the implementation is what it is

Reply:
If the Map does not have the key passed to `journeyCache` then we get an error, is that safe here or do we need to check if the key is contained in the Map?